Runtime/Triangulator.cs

```diff
@@ -396,311 +487,54 @@ private bool PointPointValidation(int i)
-                }
+                var q = positions[i] - com;
+                cov += Kron(q, q);
             }
-        }
+            cov /= positions.Length;
 
-        [BurstCompile]
-        private struct PCAInverseTransformationJob : IJobParallelForDefer
-        {
-            private NativeArray<float2> positions;
-            private NativeReference<float2>.ReadOnly comRef;
-            private NativeReference<float2>.ReadOnly scaleRef;
-            private NativeReference<float2>.ReadOnly cRef;
-            private NativeReference<float2x2>.ReadOnly URef;
+            Eigen(cov, out _, out var rotationMatrix);
 
-            public PCAInverseTransformationJob(Triangulator triangulator)
+            // Note: Taking the transpose of a rotation matrix is equivalent to taking the inverse.
+            var partialTransform = AffineTransform2D.Rotate(math.transpose(rotationMatrix)) * AffineTransform2D.Translate(-com);
+            float2 min = float.MaxValue;
+            float2 max = float.MinValue;
+            for (int i = 0; i < positions.Length; i++)
             {
-                positions = triangulator.Output.Positions.AsDeferredJobArray();
-                comRef = triangulator.com.AsReadOnly();
-                scaleRef = triangulator.scale.AsReadOnly();
-                cRef = triangulator.pcaCenter.AsReadOnly();
-                URef = triangulator.pcaMatrix.AsReadOnly();
+                var p = partialTransform.Transform(positions[i]);
+                min = math.min(p, min);
+                max = math.max(p, max);
             }
 
-            public JobHandle Schedule(Triangulator triangulator, JobHandle dependencies)
-            {
-                return this.Schedule(triangulator.Output.Positions, triangulator.Settings.BatchCount, dependencies);
-            }
+            var c = 0.5f * (min + max);
+            var s = 2f / (max - min);
 
-            public void Execute(int i)
-            {
-                var p = positions[i];
-                var com = comRef.Value;
-                var s = scaleRef.Value;
-                var c = cRef.Value;
-                var U = URef.Value;
-                positions[i] = math.mul(U, p / s + c) + com;
-            }
+            return AffineTransform2D.Scale(s) * AffineTransform2D.Translate(-c) * partialTransform;
         }
 
         [BurstCompile]
-        private struct InitialLocalTransformationJob : IJob
-        {
-            [ReadOnly]
-            private NativeArray<float2> positions;
-            private NativeReference<float2> comRef;
-            private NativeReference<float2> scaleRef;
-            private NativeList<float2> localPositions;
-
-            public InitialLocalTransformationJob(Triangulator triangulator)
+        static AffineTransform2D CalculateLocalTransformation(NativeArray<float2> positions) {
+            float2 min = float.PositiveInfinity, max = float.NegativeInfinity, com = 0;
```

Comment:
As far as I know it is safer to use `float.Min/MaxValue` here.

Reply:
As long as the positions array is not empty (which should have been checked earlier), it will not matter either way.